staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -909,6 +910,28 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		}
 	}
 
+	if len(schema.XValidations) > 0 {
+		structural, err := structuralschema.NewStructural(schema)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(fldPath.Child("x-kubernetes-validations"), err))
+		} else {
+			compResults := celcomplilation.Compile(structural)
+			if compResults.Error != nil {
+				allErrs = append(allErrs, field.InternalError(fldPath.Child("x-kubernetes-validations"), compResults.Error))
+			} else {
+				for _, cr := range compResults.Results {
+					for _, err := range cr.Errors {
+						if strings.Contains(err.Error(), "rule is not specified") {
```

Comment:
Using string comparisons is dangerous, as the message can be changed in the future and people might forget to update this check.

You should be exporting error from the cel package to use `errors.Is` or provide a function that tells you what type of error it is, based on the type or an enum stored in the error.

Reply:
Updated to have error type enum for comparison